src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py

```diff
@@ -402,6 +403,50 @@ def flexible_server_connection_string(
     }
 
 
+def connect_to_flexible_server_mysql(cmd, server_name, administrator_login, administrator_login_password, database_name=None, mysql_query=None):
+    host = '{}.mysql.database.azure.com'.format(server_name)
+    cursor = None
+    json_data = None
+    if database_name is None:
+        database_name = DEFAULT_DB_NAME
+        logger.warning("Connecting to %s database by default.", DEFAULT_DB_NAME)
+    # Connect to mysql and get cursor to run sql commands
+    try:
+        connection_kwargs = {
+            'host': host,
+            'database': database_name,
+            'user': administrator_login,
+            'password': administrator_login_password if administrator_login_password is not None else '{administrator_login_password}'
+        }
+        connection = mysql_connector.connect(**connection_kwargs)
+        logger.warning('Successfully Connected to MySQL.')
+        cursor = connection.cursor()
+    except Exception as e:
+        raise CLIError("Unable to connect to MySQL Server: {0}".format(e))
+
+    # execute query if passed in
+    if mysql_query is not None:
+        try:
+            cursor.execute(mysql_query)
+            logger.warning("Ran Database Query: '%s'", mysql_query)
+            result = cursor.fetchmany(30)  # limit to 30 rows of output for now
+            row_headers = [x[0] for x in cursor.description]  # this will extract row headers
+            # format the result for a clean display
+            json_data = []
+            for rv in result:
+                json_data.append(dict(zip(row_headers, rv)))
+        except mysql_connector.errors.Error as e:
+            raise CLIError("Unable to execute query '{0}' on MySQL Server: {1}".format(mysql_query, e))
+
+    if cursor is not None:
+        try:
+            cursor.close()
```

Comment:
You will probably not hit this code if you hit the CLI Error above(Line 439) or any other error. In that case, you will be piling up open connections. So, it would be better if you can consider having cursor.close() inside a **_finally_** block.